test/hotspot/jtreg/compiler/lib/compile_framework/CompileFramework.java

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.lib.compile_framework;
+
+import compiler.lib.compile_framework.SourceCode;
+import compiler.lib.compile_framework.CompileFrameworkException;
+import compiler.lib.compile_framework.InternalCompileFrameworkException;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.net.URL;
+import java.net.URI;
+import java.net.URLClassLoader;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import jdk.test.lib.process.ProcessTools;
+
+public class CompileFramework {
+    static final int JASM_COMPILE_TIMEOUT = 60;
+
+    private List<SourceCode> sourceCodes = new ArrayList<SourceCode>();
+    private URLClassLoader classLoader;
+
+    public void add(SourceCode sourceCode) {
+        sourceCodes.add(sourceCode);
+    }
+
+    public void printSourceCodes() {
+        for (SourceCode sourceCode : sourceCodes) {
+            System.out.println("SourceCode: " + sourceCode.className + " " + sourceCode.kind.name().toLowerCase());
+            System.out.println(sourceCode.code);
+        }
+    }
+
+    public void compile() {
+        if (classLoader != null) {
+            throw new RuntimeException("Cannot compile twice!");
+        }
+
+        printSourceCodes();
```

Comment:
Make debug printouts controllable? Fuzzing generators compiling thousands files, would generate a whole big data of those printouts.

Reply:
Good point! I'll make it dependent on -DCompileFrameworkVerbose.